db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -658,23 +685,25 @@ Status DBImpl::AtomicFlushMemTablesToOutputFiles(
       // be pessimistic and try write to a new MANIFEST.
       // TODO: distinguish between MANIFEST write and CURRENT renaming
       if (!versions_->io_status().ok()) {
-        if (total_log_size_ > 0) {
+        if (total_log_size_ > 0 || !log_io_s.ok()) {
```

Comment:
Same question here as before - when would ```versions_->io_status()``` and ```log_io_s``` both be non-ok? If ```!log_io_s.ok()```, a background error would have already been reported on line 437.